### ISSUE TITLE
The scope of a pattern variable in a catch filter should include the catch block

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#8814]
Implement pattern variables in a catch filter, whose scope should include the catch block. For example

```cs
catch (MyException ex) when (ex.Obj is int i && i == 3) {
    Console.WriteLine(i); // i is in scope here
}
```

**Comment by CyrusNajmabadi** [comment]
When fixing this, please ensure that #9074 is fixed as well.  That bug covers the actual completion experience.

**Comment by AlekseyTs** [comment]
@CyrusNajmabadi I cannot fix #9074 because it is an IDE bug. I can fix particular SemanticModel API if you provide an information about what API should be fixed and what the scenario is (i.e. what is the input to the API).

**Comment by CyrusNajmabadi** [comment]
Let me rephrase:

i should have said "please verify that #9074 is fixed.  If it is not fixed, then please reactivate it."

I basically don't want that issue potentially missed because it was duped, and then never actually addressed even though this bug was fixed.

--

Now, as to your question:  I believe starting with SemanticModel.LookupSymbols would be appropriate.  in this case if you just call it with a positoin inside the catch block, i would expect it to return that pattern variable symbol (which i assume is a LocalSymbol) in the result array.  If this happens, it's likely that completion will "just work".  If it does not, we'll have to investigate further.

**Comment by AlekseyTs** [comment]
Should be fixed now. @gafter Please verify.
